### PR TITLE
feat(repo): Added examples folder with sample Rust projects

### PR DESCRIPTION
**Pull request by 0xterminator**

### CODE REVIEW

File: examples/client/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,18 @@
+[package]
+name = "fuels-example-client"
```
Comment (Jurshsmith) [inline examples/client/Cargo.toml:2]: NIT: Can we keep the `fuel-streams` namespace?

```suggestion
name = "fuels-streams-client-examples"
```

File: examples/publisher/Cargo.toml [hunk 2]
```diff
@@ -0,0 +1,21 @@
+[package]
+name = "fuels-example-publisher"
```
Comment (Jurshsmith) [inline examples/publisher/Cargo.toml:2]:
```suggestion
name = "fuel-streams-publisher-examples"
```

File: examples/publisher/Cargo.toml [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+version = { workspace = true }
+authors = { workspace = true }
+edition = { workspace = true }
+homepage = { workspace = true }
+license = { workspace = true }
+publish = false
+repository = { workspace = true }
+description = "Fuel Data Systems Rust SDK publisher example."
```
Comment (Jurshsmith) [inline examples/publisher/Cargo.toml:10]: Nit:

```suggestion
description = "Fuel Data Systems streams publisher example."
```

File: examples/streams/Cargo.toml [hunk 4]
```diff
@@ -0,0 +1,18 @@
+[package]
+name = "fuels-example-streams"
```
Comment (Jurshsmith) [inline examples/streams/Cargo.toml:2]:
```suggestion
name = "fuel-streams-examples"
```

File: examples/publisher/Cargo.toml (hunk 2 above)
Comment (0xterminator) [reply examples/publisher/Cargo.toml]: changed, thank you!

File: examples/publisher/Cargo.toml (hunk 3 above)
Comment (0xterminator) [reply examples/publisher/Cargo.toml]: changed, thank you!

File: examples/publisher/Cargo.toml (hunk 2 above)
Comment (0xterminator) [reply examples/publisher/Cargo.toml]: changed, thank you!

**Review by Jurshsmith — DISMISSED** [review]

File: examples/streams/Cargo.toml (hunk 4 above)
Comment (0xterminator) [reply examples/streams/Cargo.toml]: changed, thx.

**Review by Jurshsmith — APPROVED** [review]

**Review by pedronauck — APPROVED** [review]

**Pull request merged**